### PR TITLE
Update to Dart SDK `1.22.0-dev.8.0`.

### PR DESCRIPTION
**Pull request by pq**
Essentially the same as https://github.com/flutter/flutter/pull/7382, updated to pull in `dev.6.0` (note: will soon be `7.1` or `8.0`) with tweaked summary generation to create spec and strong summaries (to address: https://github.com/flutter/flutter-intellij/issues/622).

The fix in https://github.com/flutter/flutter/pull/7439, should address the integration test failures last time around.

### CODE REVIEW

**Comment by pq** [comment]
@devoncarew : I'm still seeing issues with DAS in IntelliJ.  On the surface they look the same as reported in https://github.com/flutter/flutter-intellij/issues/613.  I don't suggest we land this until I get those sorted.

**Comment by pq** [comment]
Tracking DAS issues here: https://github.com/flutter/flutter-intellij/issues/622.

**Comment by pq** [comment]
**NOTE**: this cannot land until updated to point to a fresh Dart SDK that includes the fix for https://github.com/dart-lang/sdk/issues/28384.

**Comment by pq** [comment]
Updated to `dev.8.0`.  This one should be good to roll. :man_dancing: 

@devoncarew @tvolkert 

FYI @keertip

File: packages/flutter_tools/lib/src/cache.dart [hunk 1]
```diff
@@ -342,12 +346,20 @@ class FlutterEngine {
...
-    Status summaryStatus = logger.startProgress('Building Dart SDK summary...');
+    Status summarySpecStatus = logger.startProgress('Building Dart SDK spec summary...');
+    try {
+      String skyEnginePath = path.join(pkgDir.path, kSkyEngine);
+      buildSkyEngineSdkSummary(skyEnginePath, kSdkBundleSpec, false);
+    } finally {
+      summarySpecStatus.stop();
+    }
+
+    Status summaryStrongStatus = logger.startProgress('Building Dart strong SDK summary...');
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/cache.dart]: nit: `Building Dart SDK strong summary...` (to match the above)

**Comment by devoncarew** [comment]
lgtm!

**Comment by tvolkert** [comment]
LGTM!

File: packages/flutter_tools/lib/src/cache.dart (hunk 1 above)
Comment (pq) [inline packages/flutter_tools/lib/src/cache.dart]: Done!

**Pull request merged**